src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -202,14 +226,19 @@ class ActorPhaseIterator final {
     }
 
     ActorPhaseIterator& operator++() {
-        if (_iterationCheck)
+        if (_iterationCheck) {
             _iterationCheck->limitRate(_currentIteration, std::cref(*_orchestrator), _inPhase);
+            _iterationCheck->sleepAfter(std::cref(*_orchestrator), _inPhase);
```

Comment:
why is the `std::cref` necessary? Can you add a comment about it? I see type-decorators like that and it makes me think there's a hole in the type-system-design.